sentry_kafka_management/scripts/brokers.py

```diff
@@ -215,3 +219,87 @@ def remove_dynamic_configs(
         click.echo(json.dumps(error, indent=2))
         raise click.ClickException("One or more config removals failed")
     click.echo("All dynamic configs removed successfully")
+
+
+@click.command()
+@click.option(
+    "-c",
+    "--config",
+    type=click.Path(exists=True, path_type=Path),
+    required=True,
+    help="Path to the YAML configuration file",
+)
+@click.option(
+    "-n",
+    "--cluster",
+    required=True,
+    help="Name of the cluster",
+)
+@click.option(
+    "-r",
+    "--configs-record-dir",
+    type=click.Path(exists=True, path_type=Path),
+    required=True,
+    help="Path to the directory containing config change record files",
+)
+@click.option(
+    "--cleanup-records",
+    is_flag=True,
+    help="Whether to delete records in the record dir after deleting their respective configs",
+)
+@click.option(
+    "--broker-ids",
+    required=False,
+    callback=parse_broker_ids,
+    help=(
+        "Comma separated list of broker IDs to remove recorded configs from, "
+        "if not provided, recorded configs will be removed from all brokers in the cluster."
+    ),
+)
+def remove_recorded_dynamic_configs(
+    config: Path,
+    cluster: str,
+    configs_record_dir: Path,
+    cleanup_records: bool,
+    broker_ids: list[str] | None = None,
+) -> None:
+    """
+    Removes dynamic configs from a broker by reading from config record files at
+    the given path. Intended to be used to clean up configs set by the `apply-configs` script
+    that were recorded with the `--configs-record-dir` flag.
+
+    When a dynamic config is removed from a broker, the value for that config will
+    revert to being either:
+    * the static value defined in `server.properties`, if one exists
+    * the config default value, if there's no static value defined for it
+
+    Usage:
+        kafka-scripts remove-recorded-dynamic-configs -c config.yml -n my-cluster
+        --config-record-file /emergency-configs
+        --broker-ids '0,1,2'
+    """
+    yaml_config = YamlKafkaConfig(config)
+    cluster_config = yaml_config.get_clusters()[cluster]
+    client = get_admin_client(cluster_config)
+
+    configs_to_remove = read_record_dir(configs_record_dir)
```

Comment:
Question: is there meant to be multiple config files in a directory? How does one know which file a given setting is in?

Reply:
OK I understand that. However, won't this then remove all the configs in a directory? Is that the purpose of it?